integrations/pagerduty/main.py

```diff
@@ -31,7 +47,13 @@ async def on_incidents_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
         data_key=ObjectKind.INCIDENTS,
         params=query_params.generate_request_params() if query_params else None,
     ):
-        yield incidents
+        logger.info(f"Received batch with {len(incidents)} incidents")
+        tasks = [
+            enrich_incident_with_analytics_data(pager_duty_client, semaphore, incident)
+            for incident in incidents
+        ]
```

Comment:
because you are using semaphore it might take alot of time

can you somehow start yielding the incidents?

Reply:
Can you explain more on this?

Reply:
you might have 100 incidents
you are blocking the semaphore to 5 analytics requests
which means all the 100 will wait for all the analytics

can you batch request the analytics and yield a batch every time?